api/tests/integration/environments/test_clone_environment.py

```diff
@@ -0,0 +1,214 @@
+from tests.integration.helpers import get_env_feature_states_list_with_api
+
+from environments.permissions.models import UserEnvironmentPermission
+from features.models import (
+    Feature,
+    FeatureSegment,
+    FeatureState,
+    FeatureStateValue,
+)
+
+
+def test_clone_environment_returns_200(
+    admin_client, project, environment, environment_api_key
+):
+    # Given
+    env_name = "Cloned env"
+    url = reverse("api-v1:environments:environment-clone", args=[environment_api_key])
+
+    # When
+    res = admin_client.post(url, {"name": env_name})
+
+    # Then
+    assert res.status_code == status.HTTP_200_OK
+    assert res.json()["name"] == env_name
+
+
+def test_clone_environment_clones_feature_states_with_value(
+    admin_client, project, environment, environment_api_key, feature
+):
+
+    # Firstly, create an environment and update it's feature state
+
+    # Fetch the feature state id to update
+    feature_state = FeatureState.objects.get(
+        environment_id=environment, feature=feature
+    )
+    fs_update_url = reverse(
+        "api-v1:features:featurestates-detail", args=[feature_state.id]
+    )
+    data = {
+        "id": feature_state.id,
+        "feature_state_value": "new_value",
+        "enabled": False,
+        "feature": feature,
+        "environment": environment,
+        "identity": None,
+        "feature_segment": None,
+    }
+    admin_client.put(
+        fs_update_url, data=json.dumps(data), content_type="application/json"
+    )
+
+    # Now, clone the environment
+    env_name = "Cloned env"
+    url = reverse("api-v1:environments:environment-clone", args=[environment_api_key])
+    res = admin_client.post(url, {"name": env_name})
+
+    # Then, check that the clone was successful
+    assert res.status_code == status.HTTP_200_OK
+
+    # Now, fetch the feature states of the source environment
+    source_env_feature_states = get_env_feature_states_list_with_api(
+        admin_client, {"environment": environment}
+    )
+
+    # Now, fetch the feature states of the clone enviroment
+    clone_env_feature_states = get_env_feature_states_list_with_api(
+        admin_client, {"environment": res.json()["id"]}
+    )
+
+    # Finaly, compare the responses
+    assert source_env_feature_states["count"] == 1
+    assert (
+        source_env_feature_states["results"][0]["id"]
+        != clone_env_feature_states["results"][0]["id"]
+    )
+    assert (
+        source_env_feature_states["results"][0]["environment"]
+        != clone_env_feature_states["results"][0]["environment"]
+    )
+
+    assert (
+        source_env_feature_states["results"][0]["feature_state_value"]
+        == clone_env_feature_states["results"][0]["feature_state_value"]
+    )
+    assert (
+        source_env_feature_states["results"][0]["enabled"]
+        == clone_env_feature_states["results"][0]["enabled"]
+    )
+
+
+def test_clone_environment_creates_permission_with_the_current_user(
```

Comment:
I think we should keep this in it's own function, since the above test is only responsible for making sure that we clone the feature states correctly(as the name suggests).  wdyt?